sql/expression/function/date_format_test.go

```diff
@@ -0,0 +1,141 @@
+package function
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestDateFormatting(t *testing.T) {
+	dt := time.Date(2020, 2,3,4,5,6,7000, time.UTC)
+	tests := []struct {
+		formatStr string
+		expected string
+		expectErr bool
+	} {
+		{"%a", "Mon", false}, // Abbreviated weekday name (Sun to Sat)
```

Comment:
No tests that expect an error?

Reply:
Also need tests for handling nil values for date and format

Reply:
done